File: backend/src/main/java/ru/chsergeig/shoppy/service/admin/AdminOrderService.java
package ru.chsergeig.shoppy.service.admin;

import ru.chsergeig.shoppy.dto.admin.AccountDto;
import ru.chsergeig.shoppy.dto.admin.CountedGoodDto;
import ru.chsergeig.shoppy.dto.admin.GoodDto;
import ru.chsergeig.shoppy.dto.admin.OrderDto;

import java.util.List;

public interface AdminOrderService {

    List<OrderDto> getAllOrders();

    OrderDto getOrderById(Long id);

    OrderDto addOrder(String info);

    OrderDto addOrder(OrderDto dto);

    OrderDto updateOrder(OrderDto dto);

    Integer deleteOrder(Integer id);

    List<AccountDto> getAccountsByOrderId(Integer id);

    List<CountedGoodDto> getGoodsByOrderId(Integer id);

}

File: backend/src/main/kotlin/ru/chsergeig/shoppy/service/admin/AdminOrderService.kt
package ru.chsergeig.shoppy.service.admin

import ru.chsergeig.shoppy.model.AdminAccountDto
import ru.chsergeig.shoppy.model.AdminCountedGoodDto
import ru.chsergeig.shoppy.model.AdminOrderAccountsDto
import ru.chsergeig.shoppy.model.AdminOrderDto
import ru.chsergeig.shoppy.model.AdminOrderGoodsDto

interface AdminOrderService {

    fun getAllOrders(): List<AdminOrderDto?>

    fun getOrderById(
        id: Long?
    ): AdminOrderDto

    fun addOrder(
        info: String?
    ): AdminOrderDto

    fun addOrder(
        dto: AdminOrderDto?
    ): AdminOrderDto

    fun updateOrder(
        dto: AdminOrderDto?
    ): AdminOrderDto

    fun deleteOrder(
        id: Int?
    ): Int

    fun getAccountsByOrderId(
        id: Int?
    ): List<AdminAccountDto>

    fun getAccountsByOrderIds(
        requestBody: List<Int>
    ): List<AdminOrderAccountsDto>

    fun getGoodsByOrderId(
        id: Int?
    ): List<AdminCountedGoodDto>

    fun getGoodsByOrderIds(
        requestBody: List<Int>
    ): List<AdminOrderGoodsDto>
}
